docs/_data/nakadi-event-bus-api.yaml

```diff
@@ -1178,15 +1178,83 @@ paths:
         a stream.
       parameters:
         - $ref: '#/parameters/SubscriptionId'
-        - name: max_uncommitted_events
-          in: query
+        - $ref: '#/parameters/MaxUncommittedEvents'
+        - $ref: '#/parameters/BatchLimit'
+        - $ref: '#/parameters/StreamLimit'
+        - $ref: '#/parameters/BatchFlushTimeout'
+        - $ref: '#/parameters/StreamTimeout'
+        - $ref: '#/parameters/StreamKeepAliveLimit'
+        - name: X-Flow-Id
+          in: header
           description: |
-            The amount of uncommitted events Nakadi will stream before pausing the stream. When in paused
-            state and commit comes - the stream will resume. Minimal value is 1.
-          type: integer
-          format: int32
-          required: false
-          default: 10
+            The flow id of the request, which is written into the logs and passed to called services. Helpful
+            for operational troubleshooting and log analysis.
+          type: string
+      responses:
+        '200':
+          description: |
+            Ok. Stream started.
+            Stream format is a continuous series of `SubscriptionEventStreamBatch`s separated by `\n`
+          schema:
+            $ref: '#/definitions/SubscriptionEventStreamBatch'
+          headers:
+            X-Nakadi-StreamId:
+              description: |
+                the id of this stream generated by Nakadi. Must be used for committing events that were read by client
+                from this stream.
+              type: string
+        '400':
+          description: Bad Request
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Subscription not found.
+          schema:
+            $ref: '#/definitions/Problem'
+        '409':
+          description: |
+            Conflict. There are no empty slots for this subscriptions. The amount of consumers for this subscription
+            already equals the maximal value - the total amount of this subscription partitions.
+
+            This status code is also returned in the case of resetting subscription cursors request still in the
+            progress.
+          schema:
+            $ref: '#/definitions/Problem'
+        '403':
+          description: Access is forbidden for the client or event type
+          schema:
+            $ref: '#/definitions/Problem'
+
+    post:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        GET with body.
+        Starts a new stream for reading events from this subscription. Using this endpoint it is possible to specify
+        which partitions of subscription to read in this stream. If at least one of these partitions is already
+        assigned to another stream - the request will fail. It is possible to specify only the partitions that belong
```

Comment:
Do you think it would be possible that, for the same subscription, some consumers use this endpoint, and others use the rebalanced one? If that is the case, should it be that auto-rebalance consumers will be rebalanced if this calls arrives?
If both endpoints cannot be used simultaneously in the same subscription, then I think it should be stated here.

Reply:
Yes, I already thought about that problem. There are different ways to solve it. The easiest way is to forbid using both endpoints for the same subscription. 

The other option is create a new notion like assignment-type: hard-assign and soft-assign. When you specify which partitions to stream - this is hard-assign, if not - this is soft assign. The hard-assigned partitions can't be stolen, the soft-assigned partitions can be stolen. So when you connect without specifying which partitions to read - Nakadi does a rebalance of soft-assigned partitions.

Reply:
Maybe we can discuss it on a small meeting with a team.

Reply:
As we have decided we will go with the solution to forbid using hybrid mode. Also, when creating subscription the user should specify which balance mode he wants to use. I applied appropriate changes to API already.

Reply:
We know that moving from one implementation to another (blue/green) deployment is hard already. We making even harder because we are forcing consumers to create a new subscription which will have their own state(offsets).
I this setup i will be hard for the teams to switch to this new model without significant downtime and possible errors with the offset calculation leading to lost events.
For me for example, it is easy to understand how nakadi rebalance the partitions that were not requested directly.
At least, this value should be editable without deleting subscription. So users can stop their old consumers and start new from the same offsets.

Reply:
@SergKam this feature is meant a complement for the migration from low-level to high-level API. So clients using this new extension are not expected to migrate any subscriptions, since they are not using them at the moment (they are low-level users).

For the only user that is going to "migrate" from "auto-assigned" to "self-assigned" (Diga) they already confirmed that it will not be a "blue/green switch" but they are building a completely new application that will use new subscriptions.

That said, the "blue/green" deployment is a hypothetical requirement and that's why we are using this more straightforward approach. But it doesn't mean that in the future we might allow such migration, if needed.

Reply:
@SergKam I think we don't expect people to migrate from auto-balancing to custom-balancing. Auto-balancing is the way to go in 99% of cases.
This custom-balancing feature is more for those who still use Low Level API and is missing this functionality in Subscription API.

Reply:
@rcillo If we designing our API only for Diga and Low-level API Users than yes. I agree and commit.
But for me, it looks very short-sighted. Why do you think Diga needs it? This feature significantly simplifies client architecture. They can aggregate and compact data inside consumers without complex orchestration. I think many many users will use this feature as soon as it will be available and for them migration is important.

Reply:
@SergKam I think using custom balancing is actually anti-pattern of using Nakadi Subscription API.
Data Lake needs it because they want to store data by partition - the same way as it was stored in Nakadi, that's why it's very important for them to keep specific partitions streams on specific instances.

And this feature doesn't simplify client architecture because you will have to write additional code that balances partitions among your clients/instances and makes sure that all partitions are consumed etc. So it actually makes client architecture more complicated. 
So I don't think that we will promote this feature a lot. It's a specific feature for those who really need it.

Reply:
@v-stepanov one of the issues that are bound to happen is that people will use this feature and not notice when repartitioning happens, and then they will lose data. Do you think it makes sense to add a warning about that in the documentation?

Reply:
@lmontrieux I added some info regarding that:
https://github.com/zalando/nakadi/pull/834/commits/6f8d27dbbb4cbb2d8b8b438088a7c01863a37934

Reply:
> Data Lake needs it because they want to store data by partition - the same way as it was stored in Nakadi, that's why it's very important for them to keep specific partitions streams on specific instances.

I was actually looking a place where the rationale for this feature is explained. Is it anywhere aside from this comment?

Reply:
@etorreborre Good point. I added motivation to description of this PR
> Motivation for this feature is that we are deprecating Low Level API and the only feature that exists in Low Level API and is missing in Subscription API is specifying from which partition(s) to read.